client/my-sites/plans/test/main.jsx

```diff
@@ -0,0 +1,133 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import isSiteAutomatedTransferSelector from 'state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'state/sites/selectors';
+import { getIntervalTypeFromCurrentPlan } from 'state/plans/selectors';
+
+import { mapStateToProps } from '../main';
+
+jest.mock( 'lib/abtest', () => ( {
+	abtest: () => '',
+} ) );
+
+jest.mock( 'lib/analytics/index', () => ( {} ) );
+jest.mock( 'lib/analytics/ad-tracking', () => ( {} ) );
+jest.mock( 'lib/analytics/page-view-tracker', () => 'PageViewTracker' );
+jest.mock( 'lib/upgrades/actions', () => ( {} ) );
+jest.mock( 'lib/user', () => ( {} ) );
+jest.mock( 'components/main', () => 'MainComponent' );
+jest.mock( 'components/popover', () => 'Popover' );
+jest.mock( 'my-sites/plan-features', () => 'PlanFeatures' );
+jest.mock( 'state/ui/selectors', () => ( {
+	getSelectedSite: jest.fn(),
+	getSelectedSiteId: jest.fn(),
+} ) );
+
+jest.mock( 'state/selectors/is-site-automated-transfer', () => jest.fn() );
+jest.mock( 'state/plans/selectors', () => ( {
+	getIntervalTypeFromCurrentPlan: jest.fn( () => null ),
+} ) );
+jest.mock( 'state/sites/selectors', () => ( {
+	isJetpackSite: jest.fn(),
+} ) );
+
+jest.mock( 'i18n-calypso', () => ( {
+	localize: Comp => props => (
+		<Comp
+			{ ...props }
+			translate={ function( x ) {
+				return x;
+			} }
+		/>
+	),
+	numberFormat: x => x,
+} ) );
+
+const state = {};
+
+describe( 'mapStateToProps', () => {
```

Comment:
Those tests seem a bit beyond what I'd expect to see in unit tests. You can see this from the huge amount of imports you need to mock in order to test a simple logic. Perhaps add a separate selector and move the logic there, keeping it clean of all the mocks and external dependencies?

Reply:
This particular function is tied to the component so it doesn't look like a good fit to be extracted to selectors.

I don't think a large amount of mocks is inherently bad. The major reason for having a lot of mocks is that somewhere at the end of the import chain there is a top-level code doing something with `window` or some other global object or it's property. Some of those mocks are there simply in order to prevent those imports. It's this, or mocking `window.addEventListener` and such and it is true for many files containing components, and many components tests have a large number of mocked dependencies.

I understand that some selectors are already tested (not all though), but this is yet another function with it's own code, some logic operations, passing values around, and it could have it's own bugs. Multiple layers of tests are there to quickly tell you 1) if last update introduced a problem and 2) which layer is this problem on; 2 is only possible when multiple layers of code are tested.

That being said, this PR will leave this function unmodified - after implementing updates discussed with @seear there is a different strategy to deal with `intervalType`. If you insist I'll remove these tests, but personally I think that our current test coverage is pretty low and we could improve it by baby steps like this.

Reply:
I don't insist at all 😉 Just sharing my two cents.

My gut feelings are that when we need to mock so many things in order to test some logic, we are probably not isolating that logic properly. Tests for it would be way simpler if we were isolating it outside of the component. 

Also, I've noticed that tests that are dependant on many external imports and require many mocks tend to be fragile in the long term. 

I don't suggest that we remove the tests (we should totally be covering as much as possible); I only suggest isolation of the logic that we're testing, so we don't have to mock so many imports. To be honest, I don't see the benefit of not isolating the logic. It does not necessarily have to be in a selector, it could be placed to an utility function somewhere. I am a fan of separating concerns, and having to mock so many things to me is a clear factor that we are probably haven't put the logic in the right place.

The last call is your call of course.

Reply:
Actually, I want to move quickly with this PR so just removed it :-) I may issue another one where there's no hurry and we can then discuss it there